### PR TITLE
Add support for SIBR's gzip archives

### PR DESCRIPTION
**Pull request by jmaliksi**
Adds support for SIBR's gzip archives in s3 which can be downloaded from s3 with `aws --no-sign-request s3 sync s3://blaseball-archive-iliana/ ../blaseball_logs --exclude "hourly/*" --exclude "compressed-hourly/*"`

### CODE REVIEW

File: blaseballPlayback.py [hunk 1]
```diff
@@ -14,22 +15,22 @@
...
-	GREEN   = '\33[32m'
-	YELLOW  = '\33[33m'
-	BLUE    = '\33[34m'
-	VIOLET  = '\33[35m'
-	BEIGE   = '\33[36m'
-	WHITE   = '\33[37m'
-	GREY    = '\33[90m'
-	RED2    = '\33[91m'
-	GREEN2  = '\33[92m'
+	BLACK	= '\33[30m'
```
Comment (jmaliksi) [inline blaseballPlayback.py]: tbh idk what happened here 😬 trying to fix with dumb vim tricks

**Comment by Cidolfas** [comment]
I'm taking the pull request, but if I understand things right there's a potential bug introduced in this change by modifying how playback handles self.messages inside the BlaseballPlayback class.

The new code wants to mutate the state of self.messages, which could cause an issue if you connected multiple clients to the playback at the same time though SSE requests. Which is a weird use case, but technically allowed. It'll currently start a new playback stream from time 0.0 if so.

Any thoughts on how to handle that? Maybe make a self.sseClientConnected bool we can check to reject further SSE requests?

**Comment by jmaliksi** [comment]
hm, that's a good point. I'll be honest that I had a bit of trouble following the async code, so I'm not seeing what you mean by starting a new playback stream from time 0.0? Isn't that how the original recorded file does it?

**Comment by Cidolfas** [comment]
Ah, I mean that each time the playback webserver gets an SSE request it starts up an async action using the handle_sse function. That creates a response object, grabs datetime.now() for purposes of figuring out how much time has elapsed for that specific request, and then handles sending the events over the response as time passes.

Since every request starts a new async handle_sse execution the state is shared on the class instance is also shared between them. That means they'll all attempt to mutate self.messages, causing playback issues.

One thought is to block any sse requests after the first one by having handle_sse reject the request, guaranteeing that it won't lead to any issues in this weird case.

**Comment by jmaliksi** [comment]
Ah, yeah. So I think probably the correct-ish thing to do is to move the
timing logic into the Queue class, but that’d require a bigger refactor in
the SSE loop, and tbh probably not worth it for such an edge case. I’ll add
a blocker on the loop instead—should just be a matter of a bool inside a
try finally, right?

**Pull request merged**